network/p2p/unicast/manager.go

```diff
@@ -15,41 +15,61 @@ import (
 	"github.com/libp2p/go-libp2p/p2p/net/swarm"
 	"github.com/multiformats/go-multiaddr"
 	"github.com/rs/zerolog"
+	"github.com/sethvargo/go-retry"
+	"go.uber.org/atomic"
 
 	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/network/p2p"
+	"github.com/onflow/flow-go/network/p2p/unicast/protocols"
 )
 
 // MaxConnectAttemptSleepDuration is the maximum number of milliseconds to wait between attempts for a 1-1 direct connection
-const MaxConnectAttemptSleepDuration = 5
+const (
+	MaxConnectAttemptSleepDuration = 5
+
+	// DefaultRetryDelay is the default initial delay used in the exponential backoff create stream retries while
+	// waiting for dialing to peer to be complete
+	DefaultRetryDelay = 1 * time.Second
+)
+
+var (
+	_ p2p.UnicastManager = (*Manager)(nil)
+)
 
 // Manager manages libp2p stream negotiation and creation, which is utilized for unicast dispatches.
 type Manager struct {
```

Comment:
Please add metric collectors for the followings:
1. Histogram for the number of attempts on dialing a peer (labels for successful and unsuccessful dials). 
2. Histogram for the number of attempts on creating a stream to a peer (labels for successful and unsuccessful attempts). 
3. Histogram for the number of backoffs a create stream request undergoes due to dial-in progress. 
4. Histogram for the overall time it takes to dial a peer with backoff attempts. 
5. Histogram for the overall time it takes to create a stream to a peer with backoff attempts. 

Please note that you may have a single histogram capturing 1,2, and 4.

Reply:
https://github.com/onflow/flow-go/pull/3902/commits/47b11adcc579d545e01b22de6c5617e711a96621